### PR TITLE
Update pull request template with doc requirements

### PR DESCRIPTION
**Pull request by meker12**
# Description

Add release note and documentation requirements to PR template

### Fixed Issues (if relevant)

n/a

### Manual testing scenarios
n/a

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
 ### Fixed Issues (if relevant)
 <!---
-    If relevant, please provide a list of fixed issues in the format magento/ece-tools#<issue_number>.
+    If relevant, please provide a list of fixed issues in the format magento/magento-cloud-docker#<issue_number>.
```
Comment (shiftedreality) [inline .github/PULL_REQUEST_TEMPLATE.md]: Please change to `magento-cloud-components`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -16,10 +16,10 @@
...
     There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
 -->
-1. magento/magento-cloud-components#<issue_number>: Issue title
```
Comment (shiftedreality) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: Please change to `magento-cloud-components`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -30,6 +30,17 @@
 1. ...
 2. ...
 
+### Release notes
+
+For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud Docker release notes](https://devdocs.magento.com/cloud/release-notes/mcd-release-notes.html).
```
Comment (shiftedreality) [inline .github/PULL_REQUEST_TEMPLATE.md]: This must link to Cloud Components release notes

**Review by shiftedreality — CHANGES_REQUESTED** [review]

**Review by andriyShevtsov — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
    If relevant, please provide a list of fixed issues in the format magento/magento-cloud-components#<issue_number>.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -16,10 +16,10 @@
 
 ### Fixed Issues (if relevant)
 <!---
-    If relevant, please provide a list of fixed issues in the format magento/ece-tools#<issue_number>.
+    If relevant, please provide a list of fixed issues in the format magento/magento-cloud-docker#<issue_number>.
     There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
 -->
-1. magento/magento-cloud-components#<issue_number>: Issue title
+1. magento/magento-cloud-docker#<issue_number>: Issue title
```
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
1. magento/magento-cloud-components#<issue_number>: Issue title
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (meker12) [reply .github/PULL_REQUEST_TEMPLATE.md]: Those release notes don
t have too many examples, but you are correct.

Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud Docker release notes](https://devdocs.magento.com/cloud/release-notes/mcc-release-notes.html).
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -30,6 +30,17 @@
 1. ...
 2. ...
 
+### Release notes
+
+For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud Docker release notes](https://devdocs.magento.com/cloud/release-notes/mcc-release-notes.html).
```
Comment (NadiyaS) [inline .github/PULL_REQUEST_TEMPLATE.md]: [Magento Cloud **Docker** release notes] ->[Magento Cloud **Components** release notes]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -30,6 +30,17 @@
...
+
+### Associated documentation updates
+<!--
+ If your proposed update requires user documentation, submit a PR to the Magento DevDocs repository. For extensive updates requiring assistance, submit an issue to DevDocs. See https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md.
+ -->
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
+ - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
```
Comment (NadiyaS) [inline .github/PULL_REQUEST_TEMPLATE.md]: Shouldn't it be `release notes and/or documentation` ? as not always both are required

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud Component release notes](https://devdocs.magento.com/cloud/release-notes/mcc-release-notes.html).
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes or documentation changes
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -30,6 +30,17 @@
...
+
+### Associated documentation updates
+<!--
+ If your proposed update requires user documentation, submit a PR to the Magento DevDocs repository. For extensive updates requiring assistance, submit an issue to DevDocs. See https://github.com/magento/devdocs/blob/master/.github/CONTRIBUTING.md.
+ -->
+Add link to Magento DevDocs PR or Issue, if needed.
+
 ### Contribution checklist
  - [ ] Pull request has a meaningful description of its purpose
+ - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes or documentation changes
```
Comment (meker12) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (meker12) [reply .github/PULL_REQUEST_TEMPLATE.md]: Changed the wording a bit -- is it OK?

**Review by shiftedreality — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Pull request merged**